minimal_counter_dapp/README.md

```diff
@@ -0,0 +1,202 @@
+- Deploy the canister smart contract locally
+- Test backend with Candid UI and command line using DFX, and test frontend in browser 
+
+## Installation
+This example project can be cloned, installed and deployed locally, for learning and testing purposes. The instructions are based on running the example on either macOS or Linux, but when using WSL2 on Windows, the instructions will be the same.
+
+### Prerequisites
+The example project requeres the following installed:
+
+- GitHub
+- Node.js
+- DFX 
+
+GitHub and Node can be installed from various package managers. DFX can be installed following the instructions [here](https://smartcontracts.org/docs/quickstart/local-quickstart.html#download-and-install).
+
+### Install
+Install the example dapp project:
+
+```bash
+$ git clone https://github.com/dfinity/examples
+$ cd rust/minimal-counter-dapp
+$ npm install
+```
+
+## Documentation
+The three main parts of the example dapp is the backend, the Candid interface and the frontend. This example project is based on the default project, which is created when running the `dfx new --type=rust project_name`, but most of the default project code is replaced to create the counter functionality in this project.
+
+### Rust backend
+The backend functions are located in the `src/minimal_rust_dapp/lib.rs` Rust file. The backend stores the counter value, and has functions to get, increment and reset the counter value.
+
+
+#### Counter variable
+Three functions are created to make the counter work: increment(), get() and set(). The current counter value is stored as a number.
+
+
+```rust
+thread_local! {
+    static COUNTER: RefCell<candid::Nat> = RefCell::new(candid::Nat::from(0));
+}
+```
+
+#### increment()
+The `increment()` function increments the counter variable. This function is envoked when the user is clicking the button on the frontend, or when the function is called through the Candid interface.
+
+```rust
+#[update]
+fn increment() -> candid::Nat {
+    COUNTER.with(|counter| {
+        *counter.borrow_mut() += 1u64;
+    });
+
+    COUNTER.with(|counter| counter.borrow().clone())
+}
+```
+
+The function is returning the incremented counter value.
+
+#### get()
+The `get()` function returns the current counter value.
+
+```rust
+#[query]
+fn get() -> candid::Nat {
+    COUNTER.with(|counter| counter.borrow().clone())
+}
+```
+
+#### set()
+The `set()` function resets the counter value to 0 and returns the value.
+
+```rust
+#[update]
+fn set(input: candid::Nat) -> () {
+    COUNTER.with(|counter| {
+        *counter.borrow_mut() = input;
+    })
+}
+```
+
+### Candid interface
+The Candid interface is automatically created, and it has a convienient UI, which provides an easy, user friendly way to test the backend. The UI is also automatically generated, and the canister ID can be found in the `canister_ids.json` file. 
+
+The localhost version of the `canister_ids.json` file can be found in `.dfx/local/canister_ids.json` and the URL is: 
```

Comment:
Please update these instructions to use `dfx canister id`, as in the previous examples that we have worked through